content/blog/2025-11-05-valkey-helm-chart/index.md

```diff
@@ -0,0 +1,206 @@
+
+Choose an instance name for the new deployment. It must be different from the current instance to avoid overwriting resources.
+
+```bash
+export NEWINSTANCE="valkey"
+```
+
+Add the official Helm chart repository:
+
+```bash
+helm repo add valkey https://valkey.io/valkey-helm/
+helm repo update
+```
+
+Create a `values.yaml` file that matches your current deployment, for details on the configuration options, check the chart [README](https://github.com/valkey-io/valkey-helm/tree/main/valkey) and [values.yaml](https://github.com/valkey-io/valkey-helm/blob/main/valkey/values.yaml). The script below will generate a file that matches the default Bitnami Valkey configuration:
+
+**Note**: The example below provides the password as plain-text for simplicity. In production, store the password in a Kubernetes Secret and reference it using the `auth.usersExistingSecret` setting.
+
+```bash
+cat << EOF > values.yaml
+auth:
+  enabled: true
+  aclUsers:
+    default:
+      password: "$PASS"
+      permissions: "~* &* +@all"
+
+replica:
+  enabled: true
+  replicas: 3
+  persistence:
+    size: 8Gi
+
+valkeyConfig: |
+  appendonly yes
+EOF
+```
+
+Install the new Valkey instance:
+
+```bash
+helm install -n $NAMESPACE $NEWINSTANCE valkey/valkey -f values.yaml
+```
+
+Check it is running as expected:
+
+```bash
+# List new pods and ensure they are in 'Running' state
+kubectl get pods -n $NAMESPACE -l app.kubernetes.io/instance=$NEWINSTANCE
+# * Sample Output *
+# NAME       READY   STATUS    RESTARTS   AGE
+# valkey-0   1/1     Running   0          2m33s
+# valkey-1   1/1     Running   0          2m16s
+# valkey-2   1/1     Running   0          2m4s
+# valkey-3   1/1     Running   0          103s
+
+# Check that server is responding to CLI commands
+kubectl exec -n $NAMESPACE $NEWINSTANCE-0 -c valkey -- valkey-cli -a $PASS --no-auth-warning ping
+# * Sample Output *
+# PONG
+```
+
+Create a shell alias to call the Valkey CLI on the new instance:
+
+```bash
+alias new-valkey-cli="kubectl exec -n $NAMESPACE $NEWINSTANCE-0 -c valkey -- valkey-cli -a $PASS --no-auth-warning"
+```
+
+### Step 3: Enable replication
+
+Replicate data from current instance and ensure it is replicating:
+
+```bash
+# Configure password to connect to existing Valkey instance
+new-valkey-cli config set primaryauth $PASS
+# * Sample Output *
+# OK
+
+# Configure new instance to replicate data from the current instance
+new-valkey-cli replicaof $SVCPRIMARY 6379
+# * Sample Output *
+# OK
+
+# Check status of replication, it should return a 'slave' role and master_link_status as 'up'
+new-valkey-cli info | grep '^\(role\|master_host\|master_link_status\)'
+# * Sample Output *
+# role:slave
+# master_host:valkey-bitnami-primary
+# master_link_status:up
+```
+
+### Step 4: Enter maintenance window
+
+Pause all clients connecting to the Valkey server deployed using Bitnami's chart. Wait a few seconds to ensure all data is replicated and reconfigure the new instance back to primary:
```

Comment:
There is a built in command called `FAILOVER` which is the recommended way to perform this operation. https://valkey.io/commands/failover/. On newer versions users should do this.

Reply:
Thanks @madolson, I'll have to lab if authentication from the old primary to the new primary works automatically for it to become a replica. Also, whether it will cause operational issues if authentication fails. In this scenario, PAUSE CLIENT WRITE and the new primary taking the role are the important pieces, the ability of the old primary to become a replica isn't as important (in case of failure, the admin can setup replication from the new primary to the old manually and fallback).